sources/packages/web/src/services/UserAuthorizationService.ts

```diff
@@ -0,0 +1,126 @@
+import store from "@/store";
+import {
+  InstitutionRouteUserType,
+  InstitutionUserTypes,
+} from "@/types/contracts/InstitutionRouteMeta";
+import { InstitutionUserAuthRolesAndLocation } from "@/types/contracts/institution/InstitutionUser";
+export class UserAuthorizationService {
+  private static instance: UserAuthorizationService;
+
+  public static get shared(): UserAuthorizationService {
+    return this.instance || (this.instance = new this());
+  }
+
+  public isUserTypeAllowed(
+    allowedTypeList: string[],
+    urlParams?: { locationId?: string },
+    checkAllowedLocation?: { userTypes?: string[] },
+  ) {
+    const isInstitutionAdmin = store.getters["institution/myDetails"]?.isAdmin;
+    const institutionUserTypes: InstitutionUserAuthRolesAndLocation[] =
+      store.getters["institution/myAuthorizationDetails"].authorizations;
+    /* Check is user a Institution Admin and Institution 
+    Admin users has access to the requested page*/
+
+    if (isInstitutionAdmin) {
+      return allowedTypeList.some(
+        type => type === InstitutionRouteUserType.ADMIN,
+      );
+    } else {
```

Comment:
This else could be removed to simplify the logic.

Reply:
![image](https://user-images.githubusercontent.com/77353155/122455405-5c834a80-cf7a-11eb-8ef3-d5d62be4c700.png)
else added beacuse of this